From 5185202badc8d6d830069509df4d04afa8af488b Mon Sep 17 00:00:00 2001
From: Mathieu Bridon <bochecha@daitauha.fr>
Date: Wed, 24 May 2017 11:06:36 +0200
Subject: [PATCH 4/4] Rename the desktop, appdata and icon files

This uses the "reverse DNS"-like convention used by other desktop
applications and required by Flatpak.

Fixes #17
---
 install.sh                                                  |   2 +-
 playitslowly/app.py                                         |   4 ++--
 ...tslowly.appdata.xml => ch.x29a.playitslowly.appdata.xml} |   2 +-
 .../{playitslowly.desktop => ch.x29a.playitslowly.desktop}  |   2 +-
 .../apps/{playitslowly.png => ch.x29a.playitslowly.png}     | Bin
 .../apps/{playitslowly.png => ch.x29a.playitslowly.png}     | Bin
 .../apps/{playitslowly.svg => ch.x29a.playitslowly.svg}     |   0
 7 files changed, 5 insertions(+), 5 deletions(-)
 rename share/appdata/{playitslowly.appdata.xml => ch.x29a.playitslowly.appdata.xml} (90%)
 rename share/applications/{playitslowly.desktop => ch.x29a.playitslowly.desktop} (92%)
 rename share/icons/hicolor/128x128/apps/{playitslowly.png => ch.x29a.playitslowly.png} (100%)
 rename share/icons/hicolor/32x32/apps/{playitslowly.png => ch.x29a.playitslowly.png} (100%)
 rename share/icons/hicolor/scalable/apps/{playitslowly.svg => ch.x29a.playitslowly.svg} (100%)

diff --git a/install.sh b/install.sh
index be6b803..e5d8d27 100755
--- a/install.sh
+++ b/install.sh
@@ -1,4 +1,4 @@
 #!/bin/bash
 CWD=`pwd`
-pkexec bash -c "cd $CWD && python3 setup.py install && desktop-file-install /usr/local/share/applications/playitslowly.desktop"
+pkexec bash -c "cd $CWD && python3 setup.py install && desktop-file-install /usr/local/share/applications/ch.x29a.playitslowly.desktop"
 zenity --info --text "'Play it Slowly' has been installed"
diff --git a/playitslowly/app.py b/playitslowly/app.py
index f795f9a..32ad43b 100755
--- a/playitslowly/app.py
+++ b/playitslowly/app.py
@@ -95,7 +95,7 @@ class MainWindow(Gtk.Window):
         self.set_title(NAME)
 
         try:
-            self.set_icon(myGtk.iconfactory.get_icon("playitslowly", 128))
+            self.set_icon(myGtk.iconfactory.get_icon("ch.x29a.playitslowly", 128))
         except GObject.GError:
             print("could not load playitslowly icon")
 
@@ -458,7 +458,7 @@ class MainWindow(Gtk.Window):
         """show an about dialog"""
         about = Gtk.AboutDialog()
         about.set_transient_for(self)
-        about.set_logo(myGtk.iconfactory.get_icon("playitslowly", 128))
+        about.set_logo(myGtk.iconfactory.get_icon("ch.x29a.playitslowly", 128))
         about.set_name(NAME)
         about.set_program_name(NAME)
         about.set_version(VERSION)
diff --git a/share/appdata/playitslowly.appdata.xml b/share/appdata/ch.x29a.playitslowly.appdata.xml
similarity index 90%
rename from share/appdata/playitslowly.appdata.xml
rename to share/appdata/ch.x29a.playitslowly.appdata.xml
index 87ddbd8..0b754f0 100644
--- a/share/appdata/playitslowly.appdata.xml
+++ b/share/appdata/ch.x29a.playitslowly.appdata.xml
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <application>
-  <id type="desktop">playitslowly.desktop</id>
+  <id type="desktop">ch.x29a.playitslowly.desktop</id>
   <name>Play it Slowly</name>
   <summary>Play music at a different speed</summary>
   <description>
diff --git a/share/applications/playitslowly.desktop b/share/applications/ch.x29a.playitslowly.desktop
similarity index 92%
rename from share/applications/playitslowly.desktop
rename to share/applications/ch.x29a.playitslowly.desktop
index 0cf511c..bed062f 100644
--- a/share/applications/playitslowly.desktop
+++ b/share/applications/ch.x29a.playitslowly.desktop
@@ -4,7 +4,7 @@ Name=Play it Slowly
 GenericName=Play it Slowly
 Comment=A tool to play back a piece of music at a different speed or pitch
 Comment[fr]=Un outil pour lire un morceau de musique à différentes vitesses ou tonalités
-Icon=playitslowly
+Icon=ch.x29a.playitslowly
 Type=Application
 Categories=Application;AudioVideo;
 Terminal=false
diff --git a/share/icons/hicolor/128x128/apps/playitslowly.png b/share/icons/hicolor/128x128/apps/ch.x29a.playitslowly.png
similarity index 100%
rename from share/icons/hicolor/128x128/apps/playitslowly.png
rename to share/icons/hicolor/128x128/apps/ch.x29a.playitslowly.png
diff --git a/share/icons/hicolor/32x32/apps/playitslowly.png b/share/icons/hicolor/32x32/apps/ch.x29a.playitslowly.png
similarity index 100%
rename from share/icons/hicolor/32x32/apps/playitslowly.png
rename to share/icons/hicolor/32x32/apps/ch.x29a.playitslowly.png
diff --git a/share/icons/hicolor/scalable/apps/playitslowly.svg b/share/icons/hicolor/scalable/apps/ch.x29a.playitslowly.svg
similarity index 100%
rename from share/icons/hicolor/scalable/apps/playitslowly.svg
rename to share/icons/hicolor/scalable/apps/ch.x29a.playitslowly.svg
-- 
2.9.4

